docs/cookbook/prompt_engineering/text_based/self_ask.md

```diff
@@ -0,0 +1,248 @@
+            Follow up: Which state is the area code 575 located in?
+            Intermediate answer: The area code 575 is located in New Mexico.
+            Follow up: When does monsoon season end in New Mexico?
+            Intermediate answer: Monsoon season in New Mexico typically ends in mid-September.
+            So the final answer is: mid-September.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="What is the current official currency in the country where Ineabelle Diaz is a citizen?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Which country is Ineabelle Diaz a citizen of?
+            Intermediate answer: Ineabelle Diaz is from Peurto Rico, which is in the United States of America.
+            Follow up: What is the current official currency in the United States of America?
+            Intermediate answer: The current official currency in the United States is the United States dollar.
+            So the final answer is: United States dollar.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="Where was the person who founded the American Institute of Public Opinion in 1935 born?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Who founded the American Institute of Public Opinion in 1935?
+            Intermediate answer: George Gallup.
+            Follow up: Where was George Gallup born?
+            Intermediate answer: George Gallup was born in Jefferson, Iowa.
+            So the final answer is: Jefferson.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="What language is used by the director of Tiffany Memorandum?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Who directed the movie called Tiffany Memorandum?
+            Intermediate answer: Sergio Grieco.
+            Follow up: What language is used by Sergio Grieco?
+            Intermediate answer: Sergio Grieco speaks Italian.
+            So the final answer is: Italian.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="What is the sports team the person played for who scored the first touchdown in Superbowl 1?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Which player scored the first touchdown in Superbowl 1?
+            Intermediate answer: Max McGee.
+            Follow up: Which sports team did Max McGee play for?
+            Intermediate answer: Max McGee played for the Green Bay Packers.
+            So the final answer is: Green Bay Packers.
+            """
+        ),
+    ),
+]
+```
+
+Finally, we can use our Self-Ask implementation:
+
+```python
+response = self_ask_query(
```

Comment:
are we not showing standard call (no prompt engineering) on the prompt? or was it just forgotten